proposals/0000-import-shadowing.rst

```diff
@@ -0,0 +1,311 @@
+Import shadowing
+================
+
+.. author:: Gergo Erdi
+.. date-accepted:: Leave blank. This will be filled in when the proposal is accepted.
+.. ticket-url:: Leave blank. This will eventually be filled with the
+                ticket URL which will track the progress of the
+                implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/652>`_.
+.. sectnum::
+.. contents::
+
+Local binders are allowed to shadow names defined in outer
+scopes. Here, we propose that top-level binders should also be allowed
+to shadow imported names, basically treating imported modules as
+living in a scope that is "outside" the top-level one.
+
+
+Motivation
+----------
+Currently, there is no way in Haskell to shadow an imported name. The
```

Comment:
This sentence tripped me. Because you _can_ shadow an imported name, at least locally. Maybe it would be better to have the word "globally" in this sentence?

Reply:
This is completely independent feature
```haskell
-- the original is about (pseudoHaskell)
let import A (a)
    import B (b)
in  a = shadowing

-- yours is about
let import A (a)
in  let import B (a)
    in  a = shadowing
```

Reply:
> This is completely independent feature

I agree, supposing you meant this to be a reply not to this sub-thread, but @goldfirere's original top-level comment.

Reply:
:+1: to "shadow an imported name in the top-level scope"